backend/src/executors.py

```diff
@@ -15,13 +15,32 @@
 class ProtBertExecutor(Executor):
     """ProtBERT executor: https://huggingface.co/Rostlab/prot_bert"""
 
-    def __init__(self, **kwargs):
+    def __init__(self, *args, **kwargs):
         log("Initialising ProtBertExecutor.")
-        super().__init__()
+        super().__init__(*args, **kwargs)
+
+        if not os.path.isdir("./models/prot_bert") or not os.path.isdir("./tokenizers/prot_bert"):
+            ProtBertExecutor.download_model()
 
-        self.tokenizer, self.model = self.initialize_executor()
+        self.load_model()
```

Comment:
From a functional perspective, I'd rather have `self.load_model` be a pure function and return the `tokenizer` and `model` variables.

```suggestion
        self.tokenizer, self.model = self.load_model()
```

Extra: of course we can argue that `__init__` then is not pure. However that's a constructor and not really a method that you are supposed to call directly, and when you build an object the constructor does act as a pure function returning a new instantiated object (unless no static methods are involved, of course).